src/modules/72-templates-library/pages/TemplatesPage/views/TemplateListCardContextMenu/TemplateListCardContextMenu.tsx

```diff
@@ -56,6 +63,55 @@ export const TemplateListCardContextMenu: React.FC<ContextMenuProps> = (props):
     },
     [orgIdentifier, projectIdentifier, accountId, templateIdentifier]
   )
+
+  const [showDeleteTemplatesModal, hideDeleteTemplatesModal] = useModalHook(() => {
+    const content = (
+      <DeleteTemplateModal
+        template={template}
+        onClose={hideDeleteTemplatesModal}
+        onSuccess={() => {
+          hideDeleteTemplatesModal()
+        }}
+      />
+    )
+    return (
+      <Dialog enforceFocus={false} isOpen={true} className={css.deleteTemplateDialog}>
+        {isGitSyncEnabled ? <GitSyncStoreProvider>{content}</GitSyncStoreProvider> : content}
+      </Dialog>
+    )
+  }, [template, isGitSyncEnabled])
+
+  const [showTemplateSettingsModal, hideTemplateSettingsModal] = useModalHook(
+    () => (
+      <Dialog enforceFocus={false} isOpen={true} className={css.updateTemplateSettingsDialog}>
+        <TemplateSettingsModal
+          templateIdentifier={template.identifier || ''}
+          onClose={hideTemplateSettingsModal}
+          onSuccess={() => {
+            hideTemplateSettingsModal()
+          }}
+        />
+      </Dialog>
+    ),
+    [template.identifier]
+  )
+
+  const goToTemplateStudio = React.useCallback((): void => {
+    history.push(
+      routes.toTemplateStudio({
+        projectIdentifier,
+        orgIdentifier,
+        accountId,
+        module,
+        templateType: template.templateEntityType,
+        templateIdentifier: template.identifier || '',
+        versionLabel: template.versionLabel,
+        repoIdentifier: template.gitDetails?.repoIdentifier,
+        branch: template.gitDetails?.branch
+      })
+    )
+  }, [history, routes, projectIdentifier, orgIdentifier, accountId, module, template])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useCallback has an unnecessary dependency: 'routes'. Either exclude it or remove the dependency array. Outer scope values like 'routes' aren't valid dependencies because mutating them doesn't re-render the component.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9273329)
```suggestion
  }, [history, projectIdentifier, orgIdentifier, accountId, module, template])
```